### PR TITLE
 Load TransferExtensions from ServiceLoader instead of hardcoding

### PR DESCRIPTION
**Pull request by seehamrun**
#237

### CODE REVIEW

File: portability-worker/build.gradle [hunk 1]
```diff
@@ -50,7 +50,6 @@ dependencies {
     compile project(':portability-spi-service')
     compile project(':portability-spi-cloud')
     compile project(':portability-spi-transfer')
-    compile project(':extensions:data-transfer:portability-data-transfer-microsoft')
```
Comment (holachuy) [inline portability-worker/build.gradle]: Nice!

File: portability-worker/src/main/java/org/dataportabilityproject/worker/WorkerModule.java [hunk 2]
```diff
@@ -45,6 +44,22 @@
     this.context = context;
   }
 
+  private static TransferExtension findTransferExtension(
```
Comment (rtannenbaum) [inline portability-worker/src/main/java/org/dataportabilityproject/worker/WorkerModule.java]: nit: can you move this to the bottom of the file? java style guide has private methods at the bottom of classes

Comment (seehamrun) [reply portability-worker/src/main/java/org/dataportabilityproject/worker/WorkerModule.java]: Weird, the reformatter keeps moving it up there - set it manually

File: portability-worker/build.gradle (hunk 1 above)
Comment (seehamrun) [reply portability-worker/build.gradle]: Done.

**Pull request merged**